File: blockchain-analytics-flink/src/main/java/io/github/anttikaikkonen/blockchainanalyticsflink/casssandra/models/Block.java
package io.github.anttikaikkonen.blockchainanalyticsflink.casssandra.models;

import com.datastax.driver.mapping.annotations.Column;
import com.datastax.driver.mapping.annotations.PartitionKey;
import com.datastax.driver.mapping.annotations.Table;
import io.github.anttikaikkonen.bitcoinrpcclientjava.models.BlockHeader;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Table(name="block")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Block {
    
    public Block(io.github.anttikaikkonen.bitcoinrpcclientjava.models.Block block) {
        setBits(block.getBits());
        setChainwork(block.getChainwork());
        setDifficulty(block.getDifficulty());
        setHash(block.getHash());
        setHeight(block.getHeight());
        setMediantime(block.getMediantime());
        setMerkleroot(block.getMerkleroot());
        setNonce(block.getNonce());
        setPreviousblockhash(block.getPreviousblockhash());
        setTime(block.getTime());
        setVersion(block.getVersion());
        setVersionHex(block.getVersionHex());
        setSize(block.getSize());
        setTx_count(block.getTx().length);
    }
    
    public Block(BlockHeader header, int size, int tx_count) {
        setBits(header.getBits());
        setChainwork(header.getChainwork());
        setDifficulty(header.getDifficulty());
        setHash(header.getHash());
        setHeight(header.getHeight());
        setMediantime(header.getMediantime());
        setMerkleroot(header.getMerkleroot());
        setNonce(header.getNonce());
        setPreviousblockhash(header.getPreviousblockhash());
        setTime(header.getTime());
        setVersion(header.getVersion());
        setVersionHex(header.getVersionHex());
        setSize(size);
        setTx_count(tx_count);
    }
    
    @PartitionKey
    String hash;
    
    @Column(name="versionhex")
    String versionHex;
    
    int size;
    
    @Column(name="tx_count")
    int tx_count;
    
    
    int height;
    int version;
    String merkleroot;
    long time;
    long mediantime;
    long nonce;
    String bits;
    BigDecimal difficulty;
    String chainwork;
    String previousblockhash;
    
}

File: blockchain-graphql-server/src/models/block.ts
import { ObjectType, Field, Int } from "type-graphql"
import { Coin } from "./coin"

@ObjectType()
export class Block {

    @Field({ nullable: false, complexity: 1 })
    hash: string

    @Field(type => Int, { nullable: false, complexity: 1 })
    height: number

    @Field(type => Int, { nullable: false, complexity: 1 })
    size: number

    @Field(type => Int, { nullable: true, complexity: 1 })
    version: number

    @Field({ nullable: false, complexity: 1 })
    versionHex: string

    @Field({ nullable: false, complexity: 1 })
    merkleRoot: string

    @Field({ nullable: false, complexity: 1 })
    time: Date

    @Field(type => Int, { nullable: false, complexity: 1 })
    medianTime: number

    @Field(type => Int, { nullable: false, complexity: 1 })
    nonce: number

    @Field({ nullable: false, complexity: 1 })
    bits: string

    @Field({ nullable: false, complexity: 1 })
    difficulty: number

    @Field({ nullable: false, complexity: 1 })
    chainWork: string

    @Field({ nullable: false, complexity: 1 })
    previousBlockHash: string

    @Field(type => Int, { nullable: false, complexity: 1 })
    txCount: number

    @Field({ nullable: false, complexity: 1 })
    readonly coin: Coin

}